core/src/main/java/org/mule/api/lifecycle/LifecycleUtils.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.lifecycle;
+
+import org.mule.api.MuleContext;
+import org.mule.api.MuleException;
+import org.mule.api.context.MuleContextAware;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+
+public abstract class LifecycleUtils
```

Comment:
why abstract? Personal coding style... When I have a class which only has static util methods, I like to make it abstract so that it's clear that it should not be instantiated
why in this PR? It was introduced before the lifecycle changes. Back then, this class was key to manage the lifecycle. Now, it's just an utility class which comes handy some times but it's not key